sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -1338,7 +1383,7 @@ public void populateDisplayData(DisplayData.Builder builder) {
       abstract Builder<UserT, DestinationT, OutputT> setShardTemplate(
           @Nullable String shardTemplate);
 
-      abstract Builder<UserT, DestinationT, OutputT> setGenericRecords(boolean genericRecords);
+      abstract Builder<UserT, DestinationT, OutputT> setRecordClass(Class<OutputT> recordClass);
```

Comment:
This is a breaking change. It aligns better with the `Source` implementation.

Reply:
Can you make this also a separate code path?